streams/src/test/java/org/apache/kafka/streams/state/StoresTest.java

```diff
@@ -113,7 +115,7 @@ public void shouldCreateMemoryNavigableCache() {
 
     @Test
     public void shouldCreateRocksDbStore() {
-        assertThat(Stores.persistentKeyValueStore("store").get(), instanceOf(RocksDBStore.class));
+        assertThat(Stores.persistentKeyValueStore("store").get(), allOf(not(instanceOf(RocksDBTimestampedStore.class)), instanceOf(RocksDBStore.class)));
```

Comment:
We we improve this test class further (this method is ok I guess). Methods below for example:
```
    @Test
    public void shouldBuildKeyValueStore() {
        final KeyValueStore<String, String> store = Stores.keyValueStoreBuilder(
            Stores.persistentKeyValueStore("name"),
            Serdes.String(),
            Serdes.String()
        ).build();
        assertThat(store, not(nullValue()));
    }
```
Only check for not-`null` what seems to be a poor verification. (Maybe others can be improved, too).

Reply:
I agree but since the release is coming up my preference is to do this in a follow-up PR so we can get this merged sooner.

Reply:
Just to pile on other testing ideas, we can verify an exact type bound by checking assignability from both sides (since assignability is `<=` for types).

Like `store.getClass().isAssignableFrom(RocksDBStore.class)) && RocksDBStore.class.isAssignableFrom(store.getClass())`.

But I'm also in favor of doing this in a follow-up, since the fix itself looks "obviously" correct.